scipy/integrate/_ivp/common.py

```diff
@@ -8,6 +8,13 @@
 EPS = np.finfo(float).eps
 
 
+def validate_first_step(first_step):
+    """Assert that first_step is valid and return it."""
+    if first_step <= 0:
+        raise ValueError("`first_step` must be positive.")
+    return first_step
```

Comment:
Would it be worth it to add a test for the first step not being larger than `abs(t_bound - t0)`? That is also an invalid first step.

Reply:
Good suggestion.  All the current solvers as implemented will check this condition and modify the actual `first step` to satisfy it.  However, it is also best to be explicit to the user in my opinion.  This would also be useful for future solvers that allow `first_step` that may not work in this way.

I will also add a test to make sure that an exception is raised when `frist_step > abs(t_bound - t0)`